activesupport/lib/active_support/core_ext/array/access.rb

```diff
@@ -53,6 +53,17 @@ def without(*elements)
     excluding(*elements)
   end
 
+  # Returns a copy of the Array excluding the specified indiciec.
+  #
+  #   ["a", "b", "c", "d", "e", "f", "g", "h"].except_index(0, 1, -1) # => ["c", "d", "e", "f", "g"]
+  #   ["a", "b", "c", "d", "e", "f", "g", "h"].except_index(0..1, -2..-1) # => ["c", "d", "e", "f"]
+  #   ["a", "b", "c", "d", "e", "f", "g", "h"].except_index(0..1, 3, -2..-1) # => ["c", "e", "f"]
+  def except_index(*indexes)
+    indexes.each_with_object(dup) do |ind, obj|
+      ind.is_a?(Range) ? ind.each { |i| obj[i] = false } : obj[ind] = false
```

Comment:
What if we have `false`s in the Array? This would remove them

`ruby
[1,2,3,false].except_index(0) # Expected behavior: [2, 3, false] - Actual Behavior: [2, 3] 
`

Reply:
> What if we have `false`s in the Array? This would remove them
> 
> ```ruby
> [1, 2, 3, false].except_index(0)
> # Expected behavior: [2, 3, false]
> # Actual behavior: [2, 3] 
> ```

Hey, thank you!
It was due a lot of refactorings 😅